.github/workflows/build.yml

```diff
@@ -44,7 +63,20 @@ jobs:
     name: Build .war artifact
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
+      # If this is a workflow_dispatch, override RAW_URL
+    - name: Override RAW_URL for workflow_dispatch
+      if: ${{ github.event_name == 'workflow_dispatch' }}
+      run: echo "RAW_URL=${{ inputs.raw_url }}" >> $GITHUB_ENV
+      # If this workflow is called via `workflow_dispatch`, checkout the specified branch
+    - name: Git Checkout (for workflow_dispatch)
+      if: ${{ github.event_name == 'workflow_dispatch' }}
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.sha }}
+      # Otherwise, checkout the default branch
+    - name: Git Checkout
+      if: ${{ github.event_name != 'workflow_dispatch' }}
+      uses: actions/checkout@v4
```

Comment:
To avoid hacks like this, it would be better to:
1. Create one reusable workflow for building Artemis
2. Create a new workflow that has the triggers from the old build workflow and then call the reusable workflow with the proper inputs
3. Call the reusable workflow from your deployment workflow with the proper inputs

The current approach needlessly complicates the current build workflow. By doing a proper reusable workflow, it would stay about the same complexity.